src/engine/BoundsExplainer.cpp

```diff
@@ -0,0 +1,270 @@
+/*********************                                                        */
+/*! \file BoundsExplainer.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Omri Isac, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2022 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+#include "BoundsExplainer.h"
+
+BoundsExplainer::BoundsExplainer( const unsigned varsNum, const unsigned rowsNum )
+	:_varsNum( varsNum )
+	,_rowsNum( rowsNum )
+	,_upperBoundsExplanations( _varsNum, std::vector<double>( 0 ) )
+	,_lowerBoundsExplanations( _varsNum,  std::vector<double>( 0 ) )
+{
+}
+
+BoundsExplainer::~BoundsExplainer()
+{
+	for ( unsigned i = 0; i < _varsNum; ++i )
```

Comment:
No need to do any of the things that the destructor currently does. You can erase the function entirely. 
A rule of thumb: you only have to destruct stuff that was allocated using "new", with a corresponding delete. 
(An alternative is to use C++11 pointers that do this themselves - you can check with Andrew, who usually uses them).